incubator/redis-cache/templates/ss.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: "{{ template "fullname" . }}"
+  labels:
+     app: {{ template "fullname" . }}
+     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+     release: "{{ .Release.Name }}"
+     heritage: "{{ .Release.Service }}"
+spec:
+  serviceName: "{{ template "fullname" . }}"
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+        pod.beta.kubernetes.io/init-containers: '[
+           {
```

Comment:
Since you listed 1.6 as a prerequisite: How about using the new 1.6 syntax for init containers instead of the annotation approach?

See https://kubernetes.io/docs/concepts/workloads/pods/init-containers/